src/components/Auth/Auth.jsx

```diff
@@ -0,0 +1,191 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Link, withRouter } from 'react-router-dom';
+import { Button } from 'semantic-ui-react';
+import { connect } from 'react-redux';
+import { toast } from 'react-toastify';
+import AuthHeader from './AuthHeader';
+import TextInput from './TextInput';
+import { loginUser, registerUser } from '../../actions/auth/auth-dispatchers';
+import { authRef } from '../../utils/refs';
+
+/**
+ * Auth component
+ * @returns {JSX.Element} Auth component
+ */
+export class AuthComponent extends Component {
+  static propTypes = {
+    register: PropTypes.func.isRequired,
+    login: PropTypes.func.isRequired,
+    auth: PropTypes.shape().isRequired,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        authType: PropTypes.string,
+      }),
+    }).isRequired,
+  };
+
+  state = {
+    user: {
+      username: '',
+      password: '',
+      email: '',
+    },
+  };
+
+  /**
+   * @returns {undefined}
+   */
+  componentDidMount() {
+    authRef.current.addEventListener('app-toast', event => {
```

Comment:
I suggest we move registering the toast listener to the root (App) component's `componentDidMount ` method and have the window dispatch and listen for all `app-toast` events.   Also, there should be a cleanup of listeners on component unmount

```suggestion
    window.addEventListener('app-toast', event => {.......
```